akka-projection-kafka/src/main/scala/akka/kafka/scaladsl/Processor.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <https://softwaremill.com>
+ * Copyright (C) 2016 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+
+import scala.concurrent.Future
+
+import akka.NotUsed
+import akka.dispatch.ExecutionContexts
+import akka.kafka.ProducerMessage.Envelope
+import akka.kafka._
+import akka.stream.scaladsl.Keep
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+
+// TODO: copied from Alpakka Kafka PR, should we integrate this directly into projections or keep it upstream?
```

Comment:
As noted, copied from pending Alpakka Kafka PR https://github.com/akka/alpakka-kafka/pull/1090 to explore their usage, if we decide to use them.

Reply:
I did not end up using these.  I'll discuss in the upstream PR.